server/odc-service/src/main/java/com/oceanbase/odc/service/session/SessionSettingsService.java

```diff
@@ -72,16 +69,12 @@ public SessionSettings setSessionSettings(@NotNull ConnectionSession session,
                     settings.getQueryLimit(), sessionProperties.getResultSetMaxRows());
         }
         JdbcOperations jdbcOperations = session.getSyncJdbcExecutor(ConnectionSessionConstants.CONSOLE_DS_KEY);
-        String sql = "show session variables like 'autocommit'";
-        List<String> lines = jdbcOperations.query(sql, (rs, rowNum) -> rs.getString(2));
-        boolean commitFlag = false;
-        if (CollectionUtils.isNotEmpty(lines)) {
-            String autocommit = lines.get(0);
-            commitFlag = "on".equalsIgnoreCase(autocommit);
-        }
-        if (commitFlag != settings.getAutocommit()) {
-            sql = String.format("set session autocommit='%s'", settings.getAutocommit() ? "ON" : "OFF");
-            jdbcOperations.execute(sql);
+        Boolean autocommit = jdbcOperations.execute(Connection::getAutoCommit);
+        if (!autocommit.equals(settings.getAutocommit())) {
+            jdbcOperations.execute((ConnectionCallback<Void>) conn -> {
```

Comment:
we may test for oceanbase mode, the test cases can be as follow:
1. set auto commit by front end
2. execute `show variables like '%autocommit%'` to confirm if setting is working

Reply:
Passed the test following this step